src/ngMock/angular-mocks.js

```diff
@@ -2666,6 +2629,146 @@ if (window.jasmine || window.mocha) {
     }
   };
 
+  module.$$beforeAllHook = (window.before || window.beforeAll);
+  module.$$afterAllHook = (window.after || window.afterAll);
+
+  // purely for verifying this test-support functionality
+  module.$$currentSpec = function(to) {
+    if(arguments.length === 0) return to;
+    currentSpec = to;
+  };
+
+  /**
+   * @ngdoc function
+   * @name angular.mock.module.sharedInjector
+   * @description
+   *
+   * *NOTE*: This function is declared ONLY WHEN running tests with jasmine or mocha
+   *
+   * This function ensures a single injector will be used for all tests in a given describe context.
+   * This contrasts with the default behaviour where a new injector is created per test case.
+   *
+   * Use sharedInjector when you want to take advantage of Jasmine's `beforeAll()`, or mocha's
+   * `before()` methods. Call `module.sharedInjector()` before you setup any other hooks that
+   * will create (i.e call `module()`) or use (i.e call `inject()`) the injector.
+   *
+   * You cannot call `sharedInjector()` from within a context already using `sharedInjector()`.
+   *
+   * ## Example
+   *
+   * Typically beforeAll is used to make many assertions about a single operation. This can
+   * cut down test run-tims, and produce focussed tests with a single assertion.
+   *
+   *
+   * ```js
+   * describe("testing with shared injector", function() {
+   *
+   *   module.sharedInjector();
+   *
+   *   beforeAll(module("someModule"));
+   *
+   *   var object;
+   *
+   *   beforeAll(inject(function(someService) {
+   *     object = someService.getExpensiveObject();
+   *   }));
+   *
+   *   it("has calculated the answer correctly", function() {
+   *     expect(object.answer).toBe(42);
+   *   });
+   *
+   *   it("has calculated the answer within the expected time", function() {
+   *     expect(object.runTimeSeconds).toBeLessThan(10);
+   *   });
```

Comment:
Wouldn't it be more enlightening to have an example that shows multiple use of calls to `inject()`, which receive the same object - even in different `it` clauses? As it stands you could achieve this example without the PR (albeit rather ugly)...

```js
describe(..., function() {
  var object;
  it(..., inject(someService) {
    object = someService.getExpensiveObject();
    expect(object.answer).toBe(42);
  }));
  it(..., function() {
    expect(object.runTimeSeconds).toBeLessThan(10);
  });
```